lerobot/common/envs/factory.py

```diff
@@ -13,43 +13,36 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import importlib
 
 import gymnasium as gym
 from omegaconf import DictConfig
 
 
-def make_env(cfg: DictConfig, n_envs: int | None = None) -> gym.vector.VectorEnv:
+def make_env(cfg: DictConfig, n_envs: int = 1) -> gym.vector.VectorEnv:
     """Makes a gym vector environment according to the evaluation config.
 
     n_envs can be used to override eval.batch_size in the configuration. Must be at least 1.
     """
     if n_envs is not None and n_envs < 1:
         raise ValueError("`n_envs must be at least 1")
 
-    package_name = f"gym_{cfg.env.name}"
-
-    try:
-        importlib.import_module(package_name)
-    except ModuleNotFoundError as e:
-        print(
-            f"{package_name} is not installed. Please install it with `pip install 'lerobot[{cfg.env.name}]'`"
-        )
-        raise e
-
-    gym_handle = f"{package_name}/{cfg.env.task}"
+    gym_handle = f"{cfg.env.name}:{cfg.env.name}/{cfg.env.task}"
     gym_kwgs = dict(cfg.env.get("gym", {}))
+    gym_vector_kwgs = dict(cfg.env.get("gym_vector", {}))
 
     if cfg.env.get("episode_length"):
         gym_kwgs["max_episode_steps"] = cfg.env.episode_length
 
     # batched version of the env that returns an observation of shape (b, c)
-    env_cls = gym.vector.AsyncVectorEnv if cfg.eval.use_async_envs else gym.vector.SyncVectorEnv
-    env = env_cls(
-        [
-            lambda: gym.make(gym_handle, disable_env_checker=True, **gym_kwgs)
-            for _ in range(n_envs if n_envs is not None else cfg.eval.batch_size)
```

Comment:
So is eval.batch_size used exclusively for spinning up envs? I assumed it is also used in evaluation forward passes. Either way, I think it still makes more sense to name it `eval.n_envs` instead.

Reply:
Well the thing is the environment puts out a "batch" of observations and accepts a "batch" of actions. The policy just handles whatever it gets, you don't explicitly set the batch size for it.

So yeah this all comes down to variable naming now.  You may be right about the naming improvement, but it might require some chasing up to make sure documentation is up to date, and will potentially cause backwards compatibility issues for users of the library. Want to leave it for a separate PR?

Reply:
In my mind, there are two types of evaluations: (1) evaluating losses from loss functions with static validation dataset; and (2) evaluating on environment. So maybe we need both? But yes, let's leave it in a separate PR. Will revert the change.

Reply:
> In my mind, there are two types of evaluations: (1) evaluating losses from loss functions with static validation dataset; and (2) evaluating on environment. So maybe we need both? But yes, let's leave it in a separate PR. Will revert the change.

This is actually related to https://github.com/huggingface/lerobot/issues/250.

Reply:
Change reverted

Reply:
Raised an issue for this:
https://github.com/huggingface/lerobot/issues/297